File: src/utils/response.ts
export type SuccessResponse<T> = {
  success: true;
  data: T;
};

export type ErrorResponse = {
  success: false;
  error: {
    message: string;
    code?: string; 
  };
};

export function success<T>(data: T): SuccessResponse<T> {
  return {
    success: true,
    data,
  };
}

export function error(message: string, code?: string): ErrorResponse {
  return {
    success: false,
    error: { message, code },
  };
}

File: new-api/app/schemas/response.py
from typing import  Generic, TypeVar,Optional
from enum import Enum
from pydantic import BaseModel

T=TypeVar("T")

class Status(Enum):
    success= "success",
    error="error"


class Response(BaseModel,Generic[T]):
    status:Status
    message:str
    data:Optional[T]=None
    error:Optional[str]=None